src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -426,19 +427,37 @@ private static void makePublic(final CtClass c) {
         boolean foundDefault = false;
         if (!cl.isInterface()) {
           for (CtConstructor ctor : cl.getConstructors()) {
-            if (ctor.getParameterTypes().length == 0
-                && (ctor.getModifiers() & javassist.Modifier.ABSTRACT) == 0
-                && !ctor.isEmpty()) {
+            if (ctor.isEmpty()
+                && (ctor.getModifiers() & javassist.Modifier.ABSTRACT) == 0) {
               foundDefault = true;
               break;
             }
           }
         }
-        if (!foundDefault) {
+        if (!foundDefault && cl.isInterface()) {
```

Comment:
This can just be `if (cl.isInterface()) {`. `foundDefault` can never be true in this case.

Reply:
I don't think so. The loop sets `foundDefault` to true and then breaks. If no default constructor is found, `foundDefault` stays `false` throughout the loop.

Reply:
But the loop is within a `if (!cl.isInterface())`. So if `cl` _is_ an interface, then there is nothing that could set `foundDefault` to `true`?

Reply:
ah yes, you're right.

I think overall we could simply add a default ctor in cast on implementing an interface and implementing any constructor in case of an abstract class. We will use Objenesis in any case in the non-default case.